app/controllers/hiring_staff/vacancies/documents_controller.rb

```diff
@@ -52,11 +52,16 @@ def documents_form_params
 
   def redirect_if_no_supporting_documents
     supporting_documents = session[:vacancy_attributes]['supporting_documents']
-    redirect_to supporting_documents_school_job_path unless supporting_documents == 'yes'
+    redirect_to supporting_documents_school_job_path unless supporting_documents
+  end
+
+  def next_step
+    application_details_school_job_path
   end
 
   def redirect_to_next_step_if_save_and_continue
-    redirect_to application_details_school_job_path if params[:commit] == 'Save and continue'
+    vacancy = Vacancy.find(session[:vacancy_attributes]['id'])
+    redirect_to_next_step(vacancy) if params[:commit] == 'Save and continue'
```

Comment:
The documents controller does not have a next_step method currently, which next step does the application controller user in this case?

Reply:
In `develop`, the application controller never gets asked to call `next_step` by this documents controller. The next step was managed by deleted line 59 (no involvement of application controller).
With this change, `next_step` is added, so involving application controller becomes possible.